src/chainer.test.js

```
@@ -1,3 +1,36 @@
 'use strict'
 
-const { chainer } = require('./chainer');
+
+describe('cahiner', () => {
+  const { chainer } = require('./chainer');
+
+  test('should call functions from 1st to 3d', () => {
+    const f1 = jest.fn((x) => { return x * 2 });
+    const f2 = jest.fn((x) => { return x + 2 });
+    const f3 = jest.fn((x) => { return x + 2 });
+
+    const result = chainer([f1, f2, f3])(0);
+
+    expect(result).toBe(4);
+  });
+
+  test('should call functions only one time', () => {
+    const f1 = jest.fn((x) => { return x * 2 });
+    const f2 = jest.fn((x) => { return x + 2 });
+    const f3 = jest.fn((x) => { return x + 2 });
+
+    chainer([f1, f2, f3])(0);
+
+    expect(f1).toHaveBeenCalledTimes(1);
+    expect(f2).toHaveBeenCalledTimes(1);
+    expect(f3).toHaveBeenCalledTimes(1);
+  });
+
+  test('should return undefined when empty conditions', () => {
+    const f1 = jest.fn();
+    const f2 = jest.fn();
+    const f3 = jest.fn();
+
+    expect(chainer([f1, f2, f3])(0)).toBeUndefined();
+  });
+});
```

The test suite is missing required tests for:
- The empty array case (should return identity function)
- Single function in array (should behave identically to that function)
- Non-function in array (should throw a clear error)
- Falsy return values (0, '', null, false)
- Error propagation (function throws)

Consider adding these tests to fully cover the requirements.